### DISCUSSION TITLE
4 decimal places for accumulative and previous costs instead of 2?

### DISCUSSION THREAD

**Discussion by philastl** [BottlecapDave/HomeAssistant-OctopusEnergy#913] [General]
I am loving the solution you have created, I have an Octopus Mini and am obsessed with the insights that it provides.  I keep a daily, weekly, monthly check on the costs and the money generated from solar.  So much so that I am losing sleep over the points of a pence (I am from Yorkshire) and have built in a daily calculation to keep it precise.  Any chance of making the current and previous accumulative costs 4 decimal places instead of 2?

sensor.octopus_energy_electricity_**_current_accumulative_cost
sensor.octopus_energy_gas_**_current_accumulative_cost
sensor.octopus_energy_gas_**_previous_accumulative_cost
sensor.octopus_energy_electricity_**_export_previous_accumulative_cost

For example, my standing electricity charge is 0.6537, rather then 0.65.  I am having to add 0.0037 each day.  I don't want to inadvertently boil the kettle on the last day of the month and go over budget!  Just a thought but I have a workaround for the moment.

**Comment by cad42s13** [comment]
It would also be really helpful to have four decimal places on the cost attributes in the target rate sensors (they are in pounds with 2 decimal places).

The only way I've figured out to use my ohme with the agile tariff is to set a max rate in the ohme app. The target rate sensor is a brilliant way to calculate this but I need more decimal places to get the charge slots I want.

Thanks!

**Comment by BottlecapDave** [comment]
Sorry for the late response. I thought I had responded, but obviously haven't.

This is something that has gone back and forth during the development of this integration over the years depending on how the data is used or displayed by various users.

I went with two decimal places as that is usually how monetry values are displayed. The underlying calculations (e.g. total cost) is calculated with the more precise values and then rounded. I've also gone against this philosophy in places (e.g. rates) where the precision might be more important due to use (e.g. energy dashboard).

The main sensor values are easier to be more precise because the precision can now be changed in the UI to the user's liking (which wasn't possible before). The precision in the attributes are a little harder, as people would need to create template sensors or similar to get back to two decimal places. This will also opens up the question by other users of "is 4 decimal places enough". 

It's definately a situation of "damned if you do, damned if you don't", and so I need some time to ponder.

**Reply by cad42s13** [reply]
No worries, thank you very much.

I noticed that within the "Target times" attribute the "value_inc_vat" fields show the full precision of the price for that slot, so it's easy enough to figure out the max price. I do see some prices with 5 decimal places.

**Reply by BottlecapDave** [reply]
Yeah prices/rates are one of the exceptions of the above rule where they're the exact value displayed on the website. This is incase you want to use it in energy dashboard etc.

**Reply by jwt99412** [reply]
Hi Dave. I am currently extracting the rate and standing charge attributes for my tariff comparison cost sensor to give an idea of the relative rates/SC between my main tariff (tracker) and the comparison (standard variable). The rate comes out in full precision but as you say above some attributes such as the SC are only to two DP. Any chance of providing the current rates and SC as sensors in the rate comparison instance like is done in the main integration?
